tests/integration/helpers.py

```diff
@@ -485,8 +528,133 @@ def get_hostname(ops_test: OpsTest, unit_name: str) -> str:
     return unit.machine.hostname
 
 
-def get_relation(ops_test: OpsTest, relation: str = "opensearch_client"):
-    relations = ops_test.model.relations
-    for relation in relations:
-        if relation.endpoints[0].interface == "opensearch_client":
-            return relation
+# def get_relation(ops_test: OpsTest, relation: str = "opensearch_client"):
+#     relations = ops_test.model.relations
+#     for relation in relations:
+#         if relation.endpoints[0].interface == "opensearch_client":
+#             return relation
+
+
+@retry(wait=wait_fixed(wait=15), stop=stop_after_attempt(15))
+async def client_run_request(
+    ops_test,
+    unit_name: str,
+    relation: Relation,
+    method: str,
+    headers: str,
+    endpoint: str,
+    payload: str = None,
+    action: str = "run-db-request",
+    server_uri: str = "",
+):
+    # python can't have variable names with a hyphen, and Juju can't have action variables with an
+    # underscore, so this is a compromise.
+
+    app_name = unit_name.split("/")[0]
+    relation_name = [
+        relation.endpoints[i].name for i in range(2) if relation.applications[i].name == app_name
```

Comment:
:woman_facepalming: 

Blind copy-paste.

Reply:
Ah, no, it's not.

Because we need the same item of `relation.endpoints` as the item in `relation.applications`.
We need a match on the index.

`range(2)` is since we have 2 endpoints connected in a relation